vignettes/implementing-cookies.Rmd

```diff
@@ -0,0 +1,110 @@
+---
+title: "Implementing cookie consent"
+output: rmarkdown::html_vignette
+vignette: >
+  %\VignetteIndexEntry{Implementing cookie consent}
+  %\VignetteEngine{knitr::rmarkdown}
+  %\VignetteEncoding{UTF-8}
+---
+
+```{r, include = FALSE}
+knitr::opts_chunk$set(
+  collapse = TRUE,
+  comment = "#>"
+)
+```
+
+Cookie consent is a mandatory requirement for any dashboard when using cookies 
+and typically all DfE dashboards will be using cookies as part of tracking with
+Google Analytics and so will require cookie consent.
+
+> Note that to activate Google Analytics on your data dashboard, you will need 
+to contact the Statistics Development team at 
+[statistics.development@education.gov.uk](mailto:statistics.development@education.gov.uk) and request a Google Analytics key.
+
+dfeshiny provides a standard GDS compliant banner, a cookie information panel 
+and the underlying code required to allow end users to give or withhold their 
+consent for cookies to be used. If they withhold their consent, then tracking 
+with Google Analytics will be halted.
+
+Implementing cookie control on your dashboard is handled by several functions 
+in dfeshiny, with the process outlined below.
+
+The cookie control requires some underlying javascript to remove and create 
+cookies as well as switch Google Analytics on and off. This javascript needs 
+including in your dashboard's www/ folder, which can be achieved by running the 
+following command from the R console in your dashboard project / repo:
+
+```{r eval=FALSE}
+dfeshiny::init_cookies()
+```
+
+Before adding the cookie consent banner, we recommend adding your Google 
+Analytics key as a variable to your dashboard's **[global.R](https://github.com/dfe-analytical-services/dfeshiny/blob/cookie-module/tests/test_dashboard/global.R)**
+file (replacing `ABCDE12345` below with the key provided by the Statistics 
+Development team):
+
+```{r eval=FALSE}
+google_analytics_key <- "ABCDE12345"
+```
+
+You will then need to add the necessary code to your **ui.R** file to a) run the 
```

Comment:
Worth a reminder to the user to edit the "Your dashboard name" bit? Maybe seems overkill but also I have seen people overlook stuff like that!

Reply:
Fair, added.